lib/presence/stateless.js

```diff
@@ -0,0 +1,539 @@
+/*
+ * Stateless Presence
+ * ------------------
+ *
+ * This module provides an implementation of presence that works,
+ * but has some scalability problems. Each time a client joins a document,
+ * this implementation requests current presence information from all other clients,
+ * via the server. The server does not store any state at all regarding presence,
+ * it exists only in clients, hence the name "Doc Presence".
+ *
+ */
+var ShareDBError = require('../error');
+var presence = require('./index');
+var callEach = require('../util').callEach;
+
+// Check if a message represence presence.
+// Used in both ConnectionPresence and AgentPresence.
+function isPresenceMessage(data) {
+  return data.a === 'p';
+};
+
+
+/*
+ * Stateless Presence implementation of DocPresence
+ * ------------------------------------------------
+ */
+function DocPresence(doc) {
+  this.doc = doc;
+
+  // The current presence data.
+  // Map of src -> presence data
+  // Local src === ''
+  this.doc.presence = {};
+
+  // The presence objects received from the server.
+  // Map of src -> presence
+  this.received = {};
+
+  // The minimum amount of time to wait before removing processed presence from this.presence.received.
+  // The processed presence is removed to avoid leaking memory, in case peers keep connecting and disconnecting a lot.
+  // The processed presence is not removed immediately to enable avoiding race conditions, where messages with lower
+  // sequence number arrive after messages with higher sequence numbers.
+  this.receivedTimeout = 60000;
```

Comment:
Can we get some justification about why a minute is a good default amount of time? (ie is it because it's a long time compared to the "standard" lifecycle of a submit request?). How does this work if we're on a patchy connection?

Also nit: can we please write `60 * 1000`?